contrib/epee/src/net_ssl.cpp

```diff
@@ -276,20 +311,18 @@ bool ssl_handshake(boost::asio::ssl::stream<boost::asio::ip::tcp::socket> &socke
         return false;
       }
     }
-    if (!ssl_context.allow_any_cert && !ssl_context.allowed_certificates.empty() && !is_certificate_allowed(ctx, ssl_context.allowed_certificates))
-    {
+    verified = ssl_context.allow_any_cert || is_certificate_allowed(ctx, ssl_context);
+    if (!verified)
       MERROR("Certificate is not in the allowed list, connection droppped");
-      return false;
-    }
-    verified = true;
-    return true;
+
+    return verified;
```

Comment:
Did you check that this *really* works in al cases ? I remember some malarkey going on here, and this seems like a good place to have something leave a hole.